File: poo/python/Interfaces/Conta/Conta.py
#!/usr/bin/env python
#-*- coding: utf-8 -*-

"""
@author Nathan Cruz <nathan.cruz@unifesp.br>
@created 12/04/14
"""
from __future__ import unicode_literals
from __future__ import print_function

class Conta(object):

    # Atributos #
    __saldo = 0
    __numero = None


    # Construtor #
    def __init__(self, numero):
        self.__numero = numero


    def credita(self, quantia):
        if quantia > 0:
            self.__saldo += quantia
        else:
            print('Nao se pode depositar uma quantia negativa')


    def debita(self, quantia):
        if quantia > 0:
            self.__saldo -= quantia
        else:
            print('Nao se pode sacar uma quantia negativa')


    def imprime(self):
        print('Conta {0}: saldo = {1}'.format(self.getNumero(),
                                              self.getSaldo()))

    # Getters #
    def getSaldo(self):
        return self.__saldo
    def getNumero(self):
        return self.__numero

    # Setters #
    def setSaldo(self, saldo):
        self.__saldo = saldo

File: poo/java/IO/Exercicio4/Conta.java
/*
 *  Programa escrito com propósito didático apenas.
 *  @author Nathan Cruz <nathan.cruz@unifesp.br>
 *  @created 04/04/14
 *  R.A.: 86492
 */

import java.text.*;

public abstract class Conta {

    /* Atributos */
    protected double saldo;
    protected int numero;
    protected String nome;

    public void credita(double quantia) {
        if (quantia > 0){
            this.saldo += quantia;
        }
        else{
            System.err.println("Não se pode depositar uma quantia negativa");
        }
    }

    public void debita(double quantia) {
        if (quantia>0){
            this.saldo -= quantia;
        }
        else{
            System.err.println("Não se pode sacar uma quantia negativa");
        }
    }

    public abstract void atualiza(double taxa);

    /* Getters */
    public double getSaldo(){
        return this.saldo;
    }
    public double getNumero(){
        return this.numero;
    }
    public String getNome(){
        return this.nome;
    }

    public void imprime(){
        DecimalFormat casas = new DecimalFormat("#.00");
        System.out.println("Conta " + numero + ":\n" +
                           "\tTitular: " + nome +
                           "\n\tSaldo: R$" + casas.format(saldo) );
    }

    public String toString(){
        return " - Numero da Conta: "  + this.numero +
               "\n - Nome do Cliente: " + this.nome +
               "\n - Saldo: R$" + this.saldo;
    }
}
